python/pyarrow/parquet/core.py

```diff
@@ -1163,13 +1163,13 @@ def _get_pandas_index_columns(keyvalues):
 buffer_size : int, default 0
     If positive, perform read buffering when deserializing individual
     column chunks. Otherwise IO calls are unbuffered.
-partitioning : pyarrow.dataset.Partitioning or str or list of str, \
+partitioning : :py:class:`pyarrow.dataset.Partitioning` or str or list of str, \
 default "hive"
     The partitioning scheme for a partitioned dataset. The default of "hive"
     assumes directory names with key=value pairs like "/year=2009/month=11".
     In addition, a scheme like "/2009/11" is also supported, in which case
     you need to specify the field names or a full schema. See the
-    ``pyarrow.dataset.partitioning()`` function for more details."""
+    :py:func:`pyarrow.dataset.partitioning()` function for more details."""
```

Comment:
Sphinx will do the right thing here I think:  it will ignore the `()` to build the link href, but use it in the link text. So it's actually a matter of taste to include it (e.g. to more clearly flag it's a function). I just kept what was in the original here, but I'll happily remove it if that better aligns with your doc guidelines.

Reply:
actually, Sphinx will, by default, add the `()` in the link text if it's missing: https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-add_function_parentheses

so in that sense, the `()` in the `:py:func:` construct is redundant (but not invalid)